src/Xrefcheck/System.hs

```diff
@@ -50,121 +41,99 @@ askWithinCI = lookupEnv "CI" <&> \case
   Just (map C.toLower -> "true") -> True
   _ -> False
 
--- | A FilePath that has been canonicalized.
---
--- It should be created via 'canonicalizePath'.
+-- | Relative POSIX file path with some normalizations applied.
 --
--- Currently, canonical paths have been made absolute, normalised
--- regarding the running platform (e.g. Posix or Windows), with
--- indirections syntactically expanded as much as possible and
--- with no trailing path separator. All this results in a weaker
--- version than that provided by 'System.Directory.canonicalizePath'.
-newtype CanonicalPath = UnsafeCanonicalPath
-  { unCanonicalPath :: FilePath
-  } deriving newtype (Show, Eq, Ord)
-
-canonicalizePath :: FilePath -> IO CanonicalPath
-canonicalizePath = fmap canonicalize . Directory.makeAbsolute
-  where
-    canonicalize :: FilePath -> CanonicalPath
-    canonicalize = UnsafeCanonicalPath
-      . expandIndirections
-      . FP.normalise
-      . FP.dropTrailingPathSeparator
-
-    expandIndirections :: FilePath -> FilePath
-    expandIndirections = FP.joinPath
-      . reverse
-      . expand 0
-      . reverse
-      . FP.splitDirectories
-
-    expand :: Int -> [FilePath] -> [FilePath]
-    expand acc (".." : xs) = expand (acc + 1) xs
-    expand acc ("." : xs) = expand acc xs
-    expand 0 (x : xs) = x : expand 0 xs
-    expand acc (_ : xs) = expand (acc - 1) xs
-    expand acc [] = replicate acc ".."
+-- It should be created from a system dependent 'FilePath' via
+-- 'mkRelPosixLink'.
+newtype RelPosixLink = UnsafeRelPosixLink
+  { unRelPosixLink :: Text
+  } deriving newtype (Show, Eq, Ord, NFData, Buildable, Pretty)
```

Comment:
Btw, I would put it clearly somewhere in the code, that `FilePath` is used for platform-dependent filepath.

Maybe in haddock to `RelPosixPath`, for instance as "This type exists in contrast to `FilePath` which is used ..."

Reply:
Looks very good now, thanks :+1: